subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -345,7 +345,9 @@ void pe_set_state(const struct device *dev, const enum usbc_pe_state state)
 {
 	struct usbc_port_data *data = dev->data;
 
+#if !defined(CONFIG_TEST)
```

Comment:
Why remove the assert when `CONFIG_TEST` is enabled? The point behind asserts is to alert programmer about conditions that must never be false but do become false due to incorrect code (e.g. invalid API usage).

Reply:
I agree that this is not the correct thing to do but twister was complaining:
```
/home/sam/Projects/github/zephyrproject/zephyr/include/zephyr/sys/util.h:109:45: error: invalid application of 'sizeof' to incomplete type 'const struct smf_state[]'
  109 |         ((size_t) (IS_ARRAY(array) + (sizeof(array) / sizeof((array)[0]))))
      |                                             ^
/home/sam/Projects/github/zephyrproject/zephyr/include/zephyr/sys/__assert.h:116:23: note: in definition of macro '__ASSERT'
  116 |                 if (!(test)) {                                            \
      |                       ^~~~
/home/sam/Projects/github/zephyrproject/zephyr/subsys/usb/usb_c/usbc_pe_common.c:348:26: note: in expansion of macro 'ARRAY_SIZE'
  348 |         __ASSERT(state < ARRAY_SIZE(pe_states), "invalid pe_state %d", state);
      |                          ^~~~~~~~~~
```
But the build works fine.

Reply:
How odd. It seems like `const struct smf_state[]` should be completely defined regardless of `CONFIG_TEST`.

Is that newly happening in this PR?